easybuild/tools/build_from_specsfile.py

```
@@ -0,0 +1,150 @@
+                for yaml_toolchain in sw_dict[software]['toolchains']:
+                    # retrieves version number
+                    for yaml_version in sw_dict[software]['toolchains'][yaml_toolchain]['versions']:
+
+                        # if among versions there is anything else than known flags or version flag itself, it is wrong
+                        # identification of version strings is vague
+                        if str(yaml_version)[0].isdigit() \
+                                or str(yaml_version)[-1].isdigit():
+                            # creates a sw class instance
+                            try:
+                                yaml_toolchain_name = str(yaml_toolchain).split('-', 1)[0]
+                                yaml_toolchain_version = str(yaml_toolchain).split('-', 1)[1]
+                            except IndexError:
+                                yaml_toolchain_name = str(yaml_toolchain)
+                                yaml_toolchain_version = ''
+
+                            sw = SoftwareSpecs(software=software, version=yaml_version,
+                                               toolchain=yaml_toolchain, toolchain_name=yaml_toolchain_name,
+                                               toolchain_version=yaml_toolchain_version)
+
+                            # append newly created class instance to the list inside EbFromSpecs class
+                            eb.software_list.append(sw)
+
+                        elif str(yaml_version) == 'versionsuffix':
+                            try:
+                                version_info = sw_dict[software]['toolchains'][yaml_toolchain]['versions'][yaml_version]
+                                if version_info['versionsuffix'] is not None:
+                                    sw.version_suffix = version_info['versionsuffix']
+                            except (KeyError, TypeError, IndexError):
```

This looks weird, why would it be OK to silently ignore errors like this?

If `versionsuffix` may not be there, use `version_info.get('versionsuffix')`, and check for `None`?